edb/pgsql/parser/ast_builder.py

```diff
@@ -214,6 +264,144 @@ def _build_delete_stmt(n: Node, c: Context) -> pgast.DeleteStmt:
+        return pgast.SavepointStmt(savepoint_name=n["savepoint_name"])
+
+    def release(n: Node, _: Context) -> pgast.ReleaseStmt:
+        return pgast.ReleaseStmt(savepoint_name=n["savepoint_name"])
+
+    def rollback_to(n: Node, _: Context) -> pgast.RollbackToStmt:
+        return pgast.RollbackToStmt(savepoint_name=n["savepoint_name"])
+
+    def commit_prepared(n: Node, _: Context) -> pgast.CommitPreparedStmt:
+        return pgast.CommitPreparedStmt(gid=n["gid"])
+
+    def rollback_prepared(n: Node, _: Context) -> pgast.RollbackPreparedStmt:
+        return pgast.RollbackPreparedStmt(gid=n["gid"])
+
+    kinds = {
+        "BEGIN": begin,
+        "START": start,
+        "COMMIT": commit,
+        "ROLLBACK": rollback,
+        "SAVEPOINT": savepoint,
+        "RELEASE": release,
+        "ROLLBACK_TO": rollback_to,
+        "COMMIT_PREPARED": commit_prepared,
+        "ROLLBACK_PREPARED": rollback_prepared,
+    }
+
+    kind = cast(str, n["kind"]).removeprefix("TRANS_STMT_")
+    if kind not in kinds:
+        raise PSqlUnsupportedError(n)
+
+    return kinds[kind](n, c)
+
+
+def _build_transaction_options(
+    nodes: List[Node], c: Context
+) -> pgast.TransactionOptions:
+    isolation_mode = None
+    access_mode = None
+    deferrable = None
+
+    for n in nodes:
+        if "DefElem" not in n:
+            continue
+        def_e = n["DefElem"]
+        if not ("defname" in def_e or "arg" in def_e):
+            continue
+        def_name = def_e["defname"]
+        arg = _build_base_expr(def_e["arg"], c)
+
+        if def_name == "transaction_isolation":
+            if isinstance(arg, pgast.StringConstant):
+                if arg.val == "serializable":
+                    isolation_mode = pgast.IsolationMode.SERIALIZABLE
+                elif arg.val == "repeatable read":
+                    isolation_mode = pgast.IsolationMode.REPEATABLE_READ
+                elif arg.val == "read committed":
+                    isolation_mode = pgast.IsolationMode.READ_COMMITTED
+                elif arg.val == "read uncommitted":
+                    isolation_mode = pgast.IsolationMode.READ_UNCOMMITTED
+
+        elif def_name == "transaction_read_only":
+            if isinstance(arg, pgast.NumericConstant):
+                if arg.val == "1":
+                    access_mode = pgast.AccessMode.READ_ONLY
+
+        elif def_name == "transaction_deferrable":
+            if isinstance(arg, pgast.NumericConstant):
+                if arg.val == "1":
+                    deferrable = True
+
+    return pgast.TransactionOptions(
+        isolation_mode=isolation_mode or pgast.IsolationMode.READ_COMMITTED,
+        access_mode=access_mode or pgast.AccessMode.READ_WRITE,
+        deferrable=deferrable or False,
+    )
+
+
+def _build_prepare(n: Node, c: Context) -> pgast.PrepareStmt:
+    return pgast.PrepareStmt(
+        name=n["name"],
+        argtypes=_maybe_list(n, c, "params", _build_base_expr),
```

Comment:
```suggestion
        argtypes=_maybe_list(n, c, "argtypes", _build_base_expr),
```

But looks like `_build_base_expr()` doesn't support `TypeName`.